### PR TITLE
Fix CI for aws-lc-rs

### PR DESCRIPTION
**Pull request by justsmth**
### Description of changes: 
* Update the aws-lc-rs bindings generation CI check.

### Call-outs:
* Depends on: aws/aws-lc-rs#640

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

**Review by samuel40791765 — APPROVED** [review]

File: .github/workflows/aws-lc-rs.yml [hunk 1]
```diff
@@ -10,67 +10,120 @@ concurrency:
 env:
   GOPROXY: https://proxy.golang.org,direct
   AWS_LC_SYS_CMAKE_BUILDER: 1
-  RUST_NIGHTLY_TOOLCHAIN: nightly-2024-05-22
+  RUST_NIGHTLY_TOOLCHAIN: nightly
+  RUST_SCRIPT_NIGHTLY_TOOLCHAIN: nightly-2024-05-22
```
Comment (samuel40791765) [inline .github/workflows/aws-lc-rs.yml:14]: considering it's been a while, are we able to update this now?

Comment (justsmth) [reply .github/workflows/aws-lc-rs.yml:14]: We could update it, but we'd have to also change our scripts b/c they are hard-coded to this particular toolchain: https://github.com/aws/aws-lc-rs/blob/main/scripts/tools/semver.rs#L1
```
#!/usr/bin/env -S cargo +nightly-2024-05-22 -Zscript
...
```

Comment (samuel40791765) [reply .github/workflows/aws-lc-rs.yml:14]: probably in another PR then

Comment (justsmth) [reply .github/workflows/aws-lc-rs.yml:14]: Working on it now: https://github.com/aws/aws-lc-rs/pull/641

**Review by nebeid — APPROVED** [review]

**Pull request merged**